### PR TITLE
Set the binary bit on files we need to open in binary mode

### PR DESCRIPTION
**Pull request by kripken**
This might fix binaryen on windows, [which seems broken on emscripten's windows bot](http://clb.demon.fi:8112/builders/win-incoming/builds/1221/steps/Tests-default/logs/stdio)

### CODE REVIEW

**Comment by kripken** [comment]
cc @juj

File: src/support/file.h [hunk 1]
```diff
@@ -29,15 +29,15 @@
 
 namespace wasm {
 template <typename T>
-T read_file(const std::string &filename, bool debug);
+T read_file(const std::string &filename, bool binary, bool debug);
```
Comment (jfbastien) [inline src/support/file.h]: Could you use an enum instead of a flag?

Comment (kripken) [inline src/support/file.h]: It would have just 2 values though, and seem inconsistent with the next param after it, which is a bool?

Comment (jfbastien) [inline src/support/file.h]: It's more about expressing intent: what does binary=false mean? Not a big issue, I just find it easier to document it that way.

Comment (binji) [inline src/support/file.h]: It's a pretty common technique in C/C++ API design. It makes it much easier to understand the call site:
```
read_file("foo.bar", false, true); // what does this mean?
```
vs.
```
read_file("foo.bar", kAsText, kDebug);
```

**Comment by kripken** [comment]
Fair points. Ok, how about if we merge this (should unbreak a bot) and then fix both args as a followup?

**Pull request merged**